[Blockly] Regression: Get quantity from contextual info generating wrong code

## The problem

There is a blockly script. In this script there is a block looking like that:

![grafik](https://github.com/openhab/openhab-webui/assets/10487375/72dd4e9a-9394-4f54-ba88-6bff58e7226a)

The block generates the following code:

```
var newRainValue;
newRainValue = event.oldItemState !== undefined ? Quantity(event.oldItemState.toString() : undefined);
```

The last bracket is in the wrong position. That's why the script creates an error:

```
14:23:03.868 [ERROR] [internal.handler.ScriptActionHandler] - Script execution of rule with UID '871ce9c1b4' failed: org.graalvm.polyglot.PolyglotException: SyntaxError: <eval> 
   newRainValue = event.itemState !== undefined ? Quantity(event.itemState.toString() : undefined);                                                                               
                                                                                      ^                                                                                           
 <eval>:11:91 Expected , but found :
```

## Expected behavior

The code should look like this:

```
var newRainValue;
newRainValue = event.oldItemState !== undefined ? Quantity(event.oldItemState.toString()) : undefined;
```

## Steps to reproduce

1. Create a blockly script like described above.
2. Create the source code.


## Your environment

```yaml
runtimeInfo:
  version: 4.2.0.M3
  buildString: Milestone Build
locale: de-DE
systemInfo:
  configFolder: /openhab/conf
  userdataFolder: /openhab/userdata
  logFolder: /openhab/userdata/logs
  javaVersion: 17.0.11
  javaVendor: Debian
  osName: Linux
  osVersion: 6.6.28+rpt-rpi-v8
  osArchitecture: aarch64
  availableProcessors: 3
  freeMemory: 339045880
  totalMemory: 519634944
  uptime: 253131
  startLevel: 100
addons:
  - automation-jsscripting
  - binding-astro
  - binding-chromecast
  - binding-deconz
  - binding-homeconnect
  - binding-homematic
  - binding-icalendar
  - binding-logreader
  - binding-network
  - binding-ntp
  - binding-ojelectronics
  - binding-sonos
  - binding-systeminfo
  - binding-telegram
  - persistence-mapdb
  - persistence-rrd4j
  - transformation-exec
  - transformation-jsonpath
  - transformation-map
  - transformation-regex
  - transformation-scale
  - transformation-xpath
  - transformation-xslt
  - ui-basic
clientInfo:
  device:
    ios: false
    android: false
    androidChrome: false
    desktop: true
    iphone: false
    ipod: false
    ipad: false
    edge: false
    ie: false
    firefox: true
    macos: false
    windows: true
    cordova: false
    phonegap: false
    electron: false
    nwjs: false
    webView: false
    webview: false
    standalone: false
    os: windows
    pixelRatio: 1
    prefersColorScheme: light
  isSecureContext: true
  locationbarVisible: true
  menubarVisible: true
  navigator:
    cookieEnabled: true
    deviceMemory: N/A
    hardwareConcurrency: 16
    language: de
    languages:
      - de
      - en-US
      - en
    onLine: true
    platform: Win32
  screen:
    width: 2560
    height: 1440
    colorDepth: 24
  support:
    touch: false
    pointerEvents: true
    observer: true
    passiveListener: true
    gestures: false
    intersectionObserver: true
  themeOptions:
    dark: dark
    filled: true
    pageTransitionAnimation: default
    bars: light
    homeNavbar: default
    homeBackground: default
    expandableCardAnimation: default
    blocklyRenderer: geras
  userAgent: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101
    Firefox/127.0
timestamp: 2024-05-25T15:59:40.390Z

```

## Additional information

The problem occurs since 4.2.0.M3. Maybe earlier. The problem does not occur if i get string or number from contextual info. So for me i was able to solve it with a block that looks like this:

![grafik](https://github.com/openhab/openhab-webui/assets/10487375/338590de-a769-4fe5-b322-71e5f25a5739)

--- a/bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-scripts.js
+++ b/bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-scripts.js
@@ -285,7 +285,7 @@ export default function defineOHBlocks_Scripts (f7, isGraalJs, scripts) {
       if (type === 'asNumber') {
         return [`event.${contextInfo} !== undefined ? parseFloat(event.${contextInfo}.toString()) : undefined`, javascriptGenerator.ORDER_ATOMIC]
       } else if (type === 'asQuantity') {
-        return [`event.${contextInfo} !== undefined ? Quantity(event.${contextInfo}.toString() : undefined)`, javascriptGenerator.ORDER_ATOMIC]
+        return [`event.${contextInfo} !== undefined ? Quantity(event.${contextInfo}.toString()) : undefined`, javascriptGenerator.ORDER_ATOMIC]
       } else {
         return [`event.${contextInfo}?.toString()`, javascriptGenerator.ORDER_ATOMIC]
       }

